### PR TITLE
Add actor and status dropdowns to run list

### PR DESCRIPTION
**Pull request by HesterG**
Part of #25042

1. Added actor and status dropdowns first in case something is offtrack and PR is too large. 

2. Also added "No results matched." and "The workflow has no runs yet.", and  "No results matched."  will show if there is no filter results and there is no workflows (with [reference to github action](https://github.com/go-gitea/gitea/actions/workflows/files-changed.yml?query=actor%3AGiteaBot))

Demo:

https://github.com/go-gitea/gitea/assets/17645053/6e76292c-4c1f-450d-8b48-99944cfc920c

TODOs:

- [x] Get available status (same as those in `aggregateJobStatus`) instead of getting from database
- [x] Use `JOIN` to get actors, actors order by name
- [x] Make self on top

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

File: templates/repo/actions/list.tmpl [hunk 1]
```diff
@@ -19,6 +19,50 @@
...
+				<div class="ui secondary filter stackable menu gt-je">
+					<!-- Actor -->
+					<div class="ui {{if not .Actors}}disabled{{end}} dropdown jump item">
+						<span class="text">
+							{{.locale.Tr "actions.runs.actor"}}
+							{{svg "octicon-triangle-down" 14 "dropdown icon"}}
+						</span>
+						<div class="menu">
+							<div class="ui icon search input">
+								<i class="icon gt-df gt-ac gt-jc">{{svg "octicon-search" 16}}</i>
```
Comment (wxiaoguang) [inline templates/repo/actions/list.tmpl]: I guess `gt-df gt-ac gt-jc` is a no-op, at least #25163 doesn't need them, or we can review and merge #25163 first

File: templates/repo/actions/list.tmpl [hunk 2]
```diff
@@ -19,6 +19,50 @@
 				</div>
 			</div>
 			<div class="twelve wide column content">
+				<div class="ui secondary filter stackable menu gt-je">
+					<!-- Actor -->
+					<div class="ui {{if not .Actors}}disabled{{end}} dropdown jump item">
+						<span class="text">
+							{{.locale.Tr "actions.runs.actor"}}
+							{{svg "octicon-triangle-down" 14 "dropdown icon"}}
+						</span>
```
Comment (wxiaoguang) [inline templates/repo/actions/list.tmpl]: It is a wrong layout  https://fomantic-ui.com/modules/dropdown.html#dropdown

The standard layout should be `.ui.dropdown > .dropdown.icon`

File: web_src/css/repo.css [hunk 3]
```diff
@@ -1913,12 +1913,14 @@
...
+.repository.packages .empty,
+.repository.actions .empty {
   padding-top: 70px;
   padding-bottom: 100px;
 }
 
-.repository.packages .empty .svg {
+.repository.packages .empty .svg,
+.repository.actions .empty .svg {
   height: 48px;
```
Comment (wxiaoguang) [inline web_src/css/repo.css]: `{{svg "octicon-no-entry" 32}}` in code but `height: 48px;` here?

I think the `empty` CSS class doesn't make sense.

File: templates/repo/actions/list.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/actions/list.tmpl]: Yes, with just one SVG child, this is noop.

File: web_src/css/repo.css (hunk 3 above)
Comment (silverwind) [reply web_src/css/repo.css]: Yes we should render these SVGs correctly in HTML, and remove this resizing here. This approach goes for all images and SVGs, their size should be correct in HTML, making CSS resize unneeded.

Comment (HesterG) [reply web_src/css/repo.css]: This is removed https://github.com/go-gitea/gitea/pull/25118/commits/738df8f18f8515a2c3c82483300e4570d9307e57

File: templates/repo/actions/list.tmpl (hunk 2 above)
Comment (HesterG) [reply templates/repo/actions/list.tmpl]: Updated in https://github.com/go-gitea/gitea/pull/25118/commits/738df8f18f8515a2c3c82483300e4570d9307e57

File: templates/repo/actions/list.tmpl (hunk 1 above)
Comment (HesterG) [reply templates/repo/actions/list.tmpl]: Updated in https://github.com/go-gitea/gitea/pull/25118/commits/738df8f18f8515a2c3c82483300e4570d9307e57

File: templates/repo/actions/list.tmpl [hunk 4]
```diff
@@ -19,6 +19,46 @@
 				</div>
 			</div>
 			<div class="twelve wide column content">
+				<div class="ui secondary filter stackable menu gt-je">
+					<!-- Actor -->
+					<div class="ui {{if not .Actors}}disabled{{end}} dropdown jump item">
```
Comment (silverwind) [inline templates/repo/actions/list.tmpl]:
```suggestion
					<div class="ui{{if not .Actors}} disabled{{end}} dropdown jump item">
```

File: templates/repo/actions/list.tmpl [hunk 5]
```diff
@@ -19,6 +19,46 @@
...
 			</div>
 			<div class="twelve wide column content">
+				<div class="ui secondary filter stackable menu gt-je">
+					<!-- Actor -->
+					<div class="ui{{if not .Actors}} disabled{{end}} dropdown jump item">
+						<span class="text">{{.locale.Tr "actions.runs.actor"}}</span>
+						{{svg "octicon-triangle-down" 14 "dropdown icon"}}
+						<div class="menu">
+							<div class="ui icon search input">
+								<i class="icon">{{svg "octicon-search" 16}}</i>
```
Comment (silverwind) [inline templates/repo/actions/list.tmpl]:
```suggestion
								<i class="icon">{{svg "octicon-search"}}</i>
```

File: templates/repo/actions/list.tmpl [hunk 6]
```diff
@@ -19,6 +19,46 @@
...
+							{{end}}
+						</div>
+					</div>
+					<!-- Status -->
+					<div class="ui dropdown jump item">
+						<span class="text">{{.locale.Tr "actions.runs.status"}}</span>
+						{{svg "octicon-triangle-down" 14 "dropdown icon"}}
+						<div class="menu">
+							<div class="ui icon search input">
+								<i class="icon">{{svg "octicon-search" 16}}</i>
```
Comment (silverwind) [inline templates/repo/actions/list.tmpl]:
```suggestion
								<i class="icon">{{svg "octicon-search"}}</i>
```

**Review by silverwind — APPROVED** [review]

File: templates/package/shared/list.tmpl [hunk 7]
```diff
@@ -38,7 +38,7 @@
 		{{else}}
 			{{if not .HasPackages}}
 				<div class="empty center">
-					{{svg "octicon-package" 32}}
+					{{svg "octicon-package" 48}}
```
Comment (lunny) [inline templates/package/shared/list.tmpl:41]: ~unrelated change?~

**Review by wxiaoguang — APPROVED** [review]
The "MakeSelfOnTop" dependency seems strange, but it could be refactored later.
